pkg/registry/core/service/allocator/storage/storage_test.go

```diff
@@ -102,3 +102,48 @@ func TestStore(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestAllocatedButReleasedLocally(t *testing.T) {
```

Comment:
I'm not sure this is a useful test as it doesn't simulate anything real, right?  Or am I misunderstanding?

Reply:
ohh right, the bitmap locks before modifying it ... nothing should be able to modify it without updating etcd

Reply:
hmm, is testing that the local bitmap is out of date:
1. There is one port allocated in etcd
2. but that port is free in the local bitmap
3. When trying to allocate the port, if we use only the local bitmap it succeeds, that is wrong.

It checks that we are using the data from etcd for .Allocate(), is the inverse situation of the next test.